sky/clouds/kubernetes.py

```diff
@@ -532,7 +522,11 @@ def _get_image_id(resources: 'resources_lib.Resources') -> str:
         # Set environment variables for the pod. Note that SkyPilot env vars
         # are set separately when the task is run. These env vars are
         # independent of the SkyPilot task to be run.
-        k8s_env_vars = {kubernetes.IN_CLUSTER_CONTEXT_NAME_ENV_VAR: context}
+        if (skypilot_config.get_nested(('kubernetes', 'remote_identity'), None)
+                != schemas.RemoteIdentityOptions.LOCAL_CREDENTIALS.value):
+            k8s_env_vars = {kubernetes.IN_CLUSTER_CONTEXT_NAME_ENV_VAR: context}
+        else:
+            k8s_env_vars = {}
```

Comment:
nit: good to leave a comment on why we don't set the envvar
```suggestion
            # If using local credentials, the remote cluster will get the context name from the kubeconfig
            k8s_env_vars = {}
```